gadget/ondisk.go

```diff
@@ -308,22 +300,9 @@ func BuildPartitionList(dl *OnDiskVolume, pv *LaidOutVolume) (sfdiskInput *bytes
 		// Can we use the index here? Get the largest existing partition number and
 		// build from there could be safer if the disk partitions are not consecutive
 		// (can this actually happen in our images?)
-		node := deviceName(ptable.Device, pIndex)
+		node := deviceName(dl.Device, pIndex)
 		fmt.Fprintf(buf, "%s : start=%12d, size=%12d, type=%s, name=%q\n", node,
 			p.StartOffset/sectorSize, size/sectorSize, ptype, s.Name)
-
-		// Set expected labels based on role
-		switch s.Role {
```

Comment:
This ok, IIRC it was a hack to have a label be set before we create a filesystem. The helper creating FS should have used EffectiveLabel(), but we never go that far.